src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/SortedTermsWriter.java

```diff
@@ -49,9 +43,9 @@
  * <p>
```

Comment:
Thinking through one last time how we explain what's going on here...

Rather than saying terms must be added in lexicographical order, would it make more sense to say that they must be added in their unsigned byte-comparable order, given that's what the reader seems to assume?

Aside/tangent: Just in general, the use of "term" to describe the values here is almost more confusing than just calling them "blobs" or something. Maybe this was a holdover from a structure in Lucene this was based on originally or something, but what it has become is an ordered sequence of unsigned byte-comparable blobs :D

Reply:
Any terminology we **do** change here probably bears updating in the package JavaDoc thing too...

Reply:
Aside from the first comment about the "lexicographical" bit, I wouldn't worry about my tangent, to be clear. Continuing to call these "terms" is fine.

Reply:
We have two separate uses for the sorted terms. The partition keys version is only ever used for getting partition keys from row Ids. This doesn't require any order at all. The clustering keys version can look up row ids from clustering keys but only within a partition where the clustering keys are ordered. 

So, we have a partitioned case where keys must be ordered within a partition and a non-partitioned case where they don't need to be ordered at all. 

This is leading me to think that we have a case here that is similar to the `BlockPackedReader` and the `MonotonicBlockPackedReader` (not literally), where we may need to enforce different rules. In the case of the clustering keys, we should be forcing order within partitions, and the partition keys should not allow calls to the partitioned lookup at all.

I'll get these changes in. 

I have no real objection to the use of 'term' here. This is a generic package that is storing terms. If we want to stop using term we'll need to do it across SAI.